File: Kirameku-backend/admin/src/api/siteConfig.ts
import { http } from "@/utils/http";

export type SiteConfigItem = {
  id: number;
  key: string;
  value: string;
  description: string;
  updated_at: string;
};

/** 获取所有站点配置（带详情） */
export const getAllSiteConfig = () => {
  return http.request<SiteConfigItem[]>("get", "/api/site-config/list");
};

/** 获取单个配置 */
export const getSiteConfig = (key: string) => {
  return http.request<SiteConfigItem>("get", `/api/site-config/${key}`);
};

/** 更新单个配置 */
export const updateSiteConfig = (
  key: string,
  data: { value: string; description?: string }
) => {
  return http.request<SiteConfigItem>("put", `/api/site-config/${key}`, {
    data
  });
};

/** 新增配置 */
export const createSiteConfig = (data: {
  key: string;
  value: string;
  description?: string;
}) => {
  return http.request<SiteConfigItem>("post", "/api/site-config", { data });
};

/** 删除配置 */
export const deleteSiteConfig = (key: string) => {
  return http.request<{ ok: boolean }>("delete", `/api/site-config/${key}`);
};

File: Kirameku-backend/app/api/site_config.py
from fastapi import APIRouter, Depends
from pydantic import BaseModel
from sqlmodel import Session

from app.deps import get_session
from app.schemas import SiteConfigUpdate, SiteConfigOut
from app.services import site_config_service
from app.deps import get_current_user

router = APIRouter(prefix="/api/site-config", tags=["站点配置"])


class SiteConfigCreate(BaseModel):
    key: str
    value: str = ""
    description: str = ""


@router.get("")
def get_all_config(session: Session = Depends(get_session)):
    return site_config_service.get_all_config(session)


@router.get("/list")
def get_all_config_list(
    session: Session = Depends(get_session),
    _: dict = Depends(get_current_user),
):
    return site_config_service.get_all_config_list(session)


@router.get("/{key}")
def get_config(key: str, session: Session = Depends(get_session)):
    return site_config_service.get_config(session, key)


@router.post("", response_model=SiteConfigOut)
def create_config(
    data: SiteConfigCreate,
    session: Session = Depends(get_session),
    _: dict = Depends(get_current_user),
):
    return site_config_service.create_config(session, data.key, data.value, data.description)


@router.put("/{key}", response_model=SiteConfigOut)
def update_config(
    key: str,
    data: SiteConfigUpdate,
    session: Session = Depends(get_session),
    _: dict = Depends(get_current_user),
):
    return site_config_service.update_config(session, key, data)


@router.put("")
def batch_update_config(
    configs: dict,
    session: Session = Depends(get_session),
    _: dict = Depends(get_current_user),
):
    return site_config_service.batch_update_config(session, configs)


@router.delete("/{key}")
def delete_config(
    key: str,
    session: Session = Depends(get_session),
    _: dict = Depends(get_current_user),
):
    site_config_service.delete_config(session, key)
    return {"ok": True}
